CMakeLists.txt

```diff
@@ -315,6 +315,7 @@ if (OPENTURNS_HAVE_DLFCN_H)
   find_library (LIBDL_LIBRARIES NAMES dl)
   if (LIBDL_LIBRARIES)
     set (CMAKE_REQUIRED_LIBRARIES ${LIBDL_LIBRARIES})
+    list (APPEND OPENTURNS_PRIVATE_LIBRARIES ${LIBDL_LIBRARIES})
   endif ()
   set (CMAKE_REQUIRED_INCLUDES dlfcn.h)
```

Comment:
Please also remove the line above, CMAKE_REQUIRED_INCLUDES contains a list of directories, it does not make sense to set it to dlfcn.h